commit be72548ffe8618c3141c050400306b93f40a107a
Author: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Date:   Tue Dec 14 11:43:40 2010 -0500

    Revert "xen: Find an unbound irq number in reverse order (high to low)."
    
    This reverts commit 482839e7b96098f678d0404ec4dd321419ab3ea7.
    
    Conflicts:
    
    	drivers/xen/events.c

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index 0380664..8889e7f 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -401,11 +401,9 @@ static int find_unbound_irq(void)
 	int irq, res;
 	int start = get_nr_hw_irqs();
 
-	if (start == nr_irqs)
-		goto no_irqs;
 
 	/* nr_irqs is a magic value. Must not use it.*/
-	for (irq = nr_irqs-1; irq > start; irq--) {
+	for (irq = start; irq < nr_irqs; irq++) {
 		data = irq_get_irq_data(irq);
 		/* only 0->15 have init'd desc; handle irq > 16 */
 		if (!data)
@@ -414,11 +412,8 @@ static int find_unbound_irq(void)
 			break;
 		if (data->chip != &xen_dynamic_chip)
 			continue;
-		if (irq_info[irq].type == IRQT_UNBOUND)
-			return irq;
 	}
-
-	if (irq == start)
+	if (irq == nr_irqs)
 		goto no_irqs;
 
 	res = irq_alloc_desc_at(irq, -1);
